### PR TITLE
relax restrictions on anchors/aliases to closer match current YAML spec

### PR DESCRIPTION
**Pull request by Herringway**
None of the YAML specs had the existing restrictions.

### CODE REVIEW

**Comment by Herringway** [comment]
Now with proper tests and a little cleanup.

File: source/dyaml/emitter.d [hunk 1]
```diff
@@ -949,7 +950,7 @@ struct Emitter(Range, CharType) if (isOutputRange!(Range, CharType))
         ///Prepare anchor for output.
         static string prepareAnchor(const string anchor) @safe
             in(anchor != "",  "Anchor must not be empty")
-            in(anchor.all!(c => isAlphaNum(c) || c.among!('-', '_')), "Anchor contains invalid characters")
+            in(anchor.all!(c => c.isNSAnchorName), "Anchor contains invalid characters")
```
Comment (Geod24) [inline source/dyaml/emitter.d:953]:
```suggestion
            in(anchor.all!isNSAnchorName, "Anchor contains invalid characters")
```

**Review by Geod24 — APPROVED** [review]
Just one thing

File: source/dyaml/emitter.d (hunk 1 above)
Comment (Herringway) [reply source/dyaml/emitter.d]: Good catch

**Pull request merged**